### DISCUSSION TITLE
AttributeError: module 'porespy.generators' has no attribute 'spheres_from_coords'

### DISCUSSION THREAD

**Discussion by Soheil1993** [PMEAL/porespy#837] [Q&A]
Hello,

I hope you are doing well.

I have a mono-sized sphere packing in CSV format and I am following the "Generating Sphere Packings from Digital Rocks Portal" example. Unfortunately, I faced this error "AttributeError: module 'porespy.generators' has no attribute 'spheres_from_coords' ". I updated my porespy to V2.2.2 but it did not work. I do not know how should I fix it.

Any ideas and comments are appreciated!

Thanks and regards,
Soheil

**Comment by jgostick** [comment]
It just realized that we have generated the docs from the dev version, where spheres_from_coords can be found, but it's not part of an official release. Sorry about that.  Not to worry though, you can still copy&paste the code from [here](https://github.com/PMEAL/porespy/blob/dev/porespy/generators/_spheres_from_coords.py) and use it in your project.  Not sure when we'll be doing the next release of PoreSpy so hopefully this fix works for you.

**Comment by Soheil1993** [comment]
Thank you very much for your kind reply and valuable information.
I copy&paste the code that you provided. But I faced that error again!
Would you please let me know what should I use instead of "im = ps.generators.spheres_from_coords(df)" ?

**Reply by jgostick** [reply]
Hmm, when I said cut&paste you need to paste the code into the script or notebook that you're working in, then just do ``im = spheres_from_coords(df)``.  Or you can put the function in a separate file in the working directory, like 'func.py' then do ``from func import *`` (I think).  You may have to goodgle it a bit.

**Reply by Soheil1993** [reply]
Thank you very much once again. Now it works :)
